File: app/src/main/java/com/shreyaspatil/MaterialDialogExample/MainActivity.java
package com.shreyaspatil.MaterialDialogExample;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.shreyaspatil.MaterialDialog.BottomSheetMaterialDialog;
import com.shreyaspatil.MaterialDialog.MaterialDialog;
import com.shreyaspatil.MaterialDialog.interfaces.DialogInterface;
import com.shreyaspatil.MaterialDialog.interfaces.OnCancelListener;
import com.shreyaspatil.MaterialDialog.interfaces.OnDismissListener;
import com.shreyaspatil.MaterialDialog.interfaces.OnShowListener;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Animated Simple Material Dialog
        final MaterialDialog alertDialog = new MaterialDialog.Builder(this)
                .setTitle("Delete?")
                .setMessage("Are you sure want to delete this file?")
                .setCancelable(false)
                .setPositiveButton("Delete", R.drawable.ic_delete, new MaterialDialog.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        Toast.makeText(getApplicationContext(), "Deleted!", Toast.LENGTH_SHORT).show();
                        dialogInterface.dismiss();
                    }
                })
                .setNegativeButton("Cancel", R.drawable.ic_close, new MaterialDialog.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int which) {
                        Toast.makeText(getApplicationContext(), "Cancelled!", Toast.LENGTH_SHORT).show();
                        dialogInterface.dismiss();
                    }
                })
                .setAnimation("delete_anim.json")
                .build();

        // Animated BottomSheet Material Dialog
        final BottomSheetMaterialDialog sheetDialog = new BottomSheetMaterialDialog.Builder(this)
                .setTitle("Delete?")
                .setMessage("Are you sure want to delete this file?")
                .setCancelable(false)
                .setPositiveButton("Delete", R.drawable.ic_delete, new BottomSheetMaterialDialog.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        Toast.makeText(getApplicationContext(), "Deleted!", Toast.LENGTH_SHORT).show();
                        dialogInterface.dismiss();
                    }
                })
                .setNegativeButton("Cancel", R.drawable.ic_close, new BottomSheetMaterialDialog.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int which) {
                        Toast.makeText(getApplicationContext(), "Cancelled!", Toast.LENGTH_SHORT).show();
                        dialogInterface.dismiss();
                    }
                })
                .setAnimation("delete_anim.json")
                .build();

        findViewById(R.id.button_smdialog).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                alertDialog.show();
            }
        });

        findViewById(R.id.button_bsdialog).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sheetDialog.show();
            }
        });

        sheetDialog.setOnCancelListener(new OnCancelListener() {
            @Override
            public void onCancel(DialogInterface dialogInterface) {
                System.out.println("CANCELLED");
            }
        });

        sheetDialog.setOnShowListener(new OnShowListener() {
            @Override
            public void onShow(DialogInterface dialogInterface) {
                System.out.println("SHOWED");
            }
        });

        sheetDialog.setOnDismissListener(new OnDismissListener() {
            @Override
            public void onDismiss(DialogInterface dialogInterface) {
                System.out.println("DISMISS");
            }
        });
    }
}

File: app/src/main/java/com/shreyaspatil/MaterialDialogExample/MainActivity.kt
package com.shreyaspatil.MaterialDialogExample

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.shreyaspatil.MaterialDialog.BottomSheetMaterialDialog
import com.shreyaspatil.MaterialDialog.MaterialDialog
import com.shreyaspatil.MaterialDialog.bottomSheetMaterialDialog
import com.shreyaspatil.MaterialDialog.interfaces.DialogInterface
import com.shreyaspatil.MaterialDialog.interfaces.OnCancelListener
import com.shreyaspatil.MaterialDialog.interfaces.OnDismissListener
import com.shreyaspatil.MaterialDialog.interfaces.OnShowListener
import com.shreyaspatil.MaterialDialog.materialDialog
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener, OnShowListener, OnCancelListener, OnDismissListener {

    private lateinit var mSimpleDialog: MaterialDialog
    private lateinit var mAnimatedDialog: MaterialDialog
    private lateinit var mSimpleBottomSheetDialog: BottomSheetMaterialDialog
    private lateinit var mAnimatedBottomSheetDialog: BottomSheetMaterialDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mSimpleDialog = materialDialog(this) {
            title = "Delete"
            message = "Are you sure you want to delete this file?"
            isCancelable = false
            setPositiveButton("Delete", R.drawable.ic_delete) { dialog, which ->
                Toast.makeText(applicationContext, "Deleted!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setNegativeButton("Cancel", R.drawable.ic_close) { dialog, which ->
                Toast.makeText(applicationContext, "Cancelled!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
        }

        mSimpleBottomSheetDialog = bottomSheetMaterialDialog(this) {
            title = "Delete"
            message = "Are you sure you want to delete this file?"
            isCancelable = false
            setPositiveButton("Delete", R.drawable.ic_delete) { dialog, which ->
                Toast.makeText(applicationContext, "Deleted!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setNegativeButton("Cancel", R.drawable.ic_close) { dialog, which ->
                Toast.makeText(applicationContext, "Cancelled!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
        }

        mAnimatedDialog = materialDialog(this) {
            title = "Delete"
            message = "Are you sure you want to delete this file?"
            isCancelable = false
            setPositiveButton("Delete", R.drawable.ic_delete) { dialog, which ->
                Toast.makeText(applicationContext, "Deleted!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setNegativeButton("Cancel", R.drawable.ic_close) { dialog, which ->
                Toast.makeText(applicationContext, "Cancelled!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setAnimation("delete_anim.json")
        }

        mAnimatedBottomSheetDialog = bottomSheetMaterialDialog(this) {
            title = "Delete"
            message = "Are you sure you want to delete this file?"
            isCancelable = false
            setPositiveButton("Delete", R.drawable.ic_delete) { dialog, which ->
                Toast.makeText(applicationContext, "Deleted!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setNegativeButton("Cancel", R.drawable.ic_close) { dialog, which ->
                Toast.makeText(applicationContext, "Cancelled!", Toast.LENGTH_SHORT).show()
                dialog.dismiss()
            }
            setAnimation("delete_anim.json")
        }


        button_simple_dialog.setOnClickListener(this)
        button_simple_bottomsheet_dialog.setOnClickListener(this)
        button_animated_dialog.setOnClickListener(this)
        button_animated_bottomsheet_dialog.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.button_simple_dialog -> mSimpleDialog.show()
            R.id.button_simple_bottomsheet_dialog -> mSimpleBottomSheetDialog.show()
            R.id.button_animated_dialog -> mAnimatedDialog.show()
            R.id.button_animated_bottomsheet_dialog -> mAnimatedBottomSheetDialog.show()
        }
    }

    override fun onShow(dialogInterface: DialogInterface) {

    }

    override fun onCancel(dialogInterface: DialogInterface) {

    }

    override fun onDismiss(dialogInterface: DialogInterface) {

    }
}